components/perf/dynamicRendering/src/lazyContent.js

```diff
@@ -1,73 +1,22 @@
-import React, {Component} from 'react'
+import React from 'react'
 import PropTypes from 'prop-types'
-
-export class LazyContent extends Component {
-  constructor(props) {
-    super(props)
-    this.refTarget = React.createRef()
-  }
-
-  _observer = null
-
-  state = {
-    isIntersecting: false
-  }
-
-  handleChange = ([{isIntersecting}]) => {
-    if (isIntersecting) {
-      this.unobserve()
-      this._observer = null
-      this.setState({isIntersecting})
-    }
-  }
-
-  unobserve = () => {
-    try {
-      this._observer.unobserve(this.refTarget.current)
-    } catch (e) {
-      console.warn(e)
-    }
-  }
-
-  _startIntersectionObserver = () => {
-    this._observer = new window.IntersectionObserver(this.handleChange)
-    try {
-      this._observer.observe(this.refTarget.current, {
-        rootMargin: this.props.rootMargin
-      })
-    } catch (e) {
-      console.warn(e)
-    }
+import {useNearScreen} from '@schibstedspain/sui-react-hooks'
+
+export default function LazyContent({
+  children,
+  rootMargin = '100px 0px 0px',
+  placeholder,
+  height,
+}) {
+  const [isNearScreen, fromRef] = useNearScreen({offset: rootMargin})
+
+  if (isNearScreen) {
+    return children
+  } else if (placeholder) {
+    return <div ref={fromRef}>{placeholder}</div>
+  } else {
+    return <div ref={fromRef} style={{height: `${height}px`}} />
```

Comment:
I think that we need this hack here: `marginBottom = 1px`

Reply:
The hack is already here: https://github.com/SUI-Components/schibsted-spain-components/pull/142/files#diff-11ae61defc028aa316a9a02f88c3c488R54